PWGCF/Flow/Tasks/resonancesGfwFlow.cxx

```diff
@@ -0,0 +1,341 @@
+  {
+    if (!collision.sel8() || !collision.selection_bit(aod::evsel::kNoTimeFrameBorder) || !collision.selection_bit(aod::evsel::kNoITSROFrameBorder) || !collision.selection_bit(aod::evsel::kNoSameBunchPileup) || !collision.selection_bit(aod::evsel::kIsGoodZvtxFT0vsPV) || !collision.selection_bit(o2::aod::evsel::kNoCollInTimeRangeStandard)) {
+      return;
+    }
+
+    const auto cent = collision.centFT0C();
+    int nTot = tracks.size();
+    float vtxz = collision.posZ();
+    int occupancy = collision.trackOccupancyInTimeRange();
+
+    if (occupancy > cfgCutOccupancy) {
+      return;
+    }
+
+    if (additionalEvsel && !eventSelected(collision, cent)) {
+      return;
+    }
+
+    histos.fill(HIST("hVtxZ"), vtxz);
+    histos.fill(HIST("hMult"), nTot);
+    histos.fill(HIST("hCent"), cent);
+
+    auto posSlicedTracks = posTracks->sliceByCached(aod::track::collisionId, collision.globalIndex(), cache);
+    auto negSlicedTracks = negTracks->sliceByCached(aod::track::collisionId, collision.globalIndex(), cache);
+
+    for (auto const& track1 : posSlicedTracks) {
+      // track selection
+      if (!selectionTrack(track1)) {
+        continue;
+      }
+      // PID check
+      if (ispTdepPID && !selectionPIDpTdependent(track1)) {
+        continue;
+      }
+      if (!ispTdepPID && !selectionPID(track1)) {
+        continue;
+      }
+      histos.fill(HIST("KaplusTPC"), track1.pt(), track1.tpcNSigmaKa());
+      histos.fill(HIST("KaplusTOF"), track1.pt(), track1.tofNSigmaKa());
+      auto track1ID = track1.globalIndex();
+
+      for (auto const& track2 : negSlicedTracks) {
+        // track selection
+        if (!selectionTrack(track2)) {
+          continue;
+        }
+        // PID check
+        if (ispTdepPID && !selectionPIDpTdependent(track2)) {
+          continue;
+        }
+        if (!ispTdepPID && !selectionPID(track2)) {
+          continue;
+        }
+        auto track2ID = track2.globalIndex();
+        if (track2ID == track1ID) {
+          continue;
+        }
+        if (!selectionPair(track1, track2)) {
+          continue;
+        }
+        if (removefaketrack && isFakeKaon(track1)) {
+          continue;
+        }
+        if (removefaketrack && isFakeKaon(track2)) {
+          continue;
+        }
+        histos.fill(HIST("KaminusTPC"), track2.pt(), track2.tpcNSigmaKa());
+        histos.fill(HIST("KaminusTOF"), track2.pt(), track2.tofNSigmaKa());
+        kaonPlus = ROOT::Math::PxPyPzMVector(track1.px(), track1.py(), track1.pz(), massKaPlus);
+        kaonminus = ROOT::Math::PxPyPzMVector(track2.px(), track2.py(), track2.pz(), massKaPlus);
+        phiMom = kaonPlus + kaonminus;
+        if (std::abs(phiMom.Rapidity()) < confRapidity) {
+          histos.fill(HIST("hPhiMass_sparse"), phiMom.M(), phiMom.Pt(), cent);
+          histos.fill(HIST("hPhi"), phiMom.Phi());
+          histos.fill(HIST("hEta"), phiMom.Eta());
+        }
```

Comment:
Consider that this processing for `track2` is repeated again for each `track1`
This has an evident effect on the singles histograms content, but also in the execution efficiency
- each track is selected n times
- each track is identified n times
- each track is checked as fake kaon n times
Have it in mind if you have execution warnings in hyperloop